tests/test-amp-cache-utilities.php

```diff
@@ -0,0 +1,509 @@
+				),
+				'/c/s/example.com/path/to/resource.ext',
+			),
+			'https_content_specify_scheme_success' => array(
+				array(
+					'url' => 'https://example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'https',
+				),
+				'/c/s/example.com/path/to/resource.ext',
+			),
+			'https_content_specify_different_scheme_success' => array(
+				array(
+					'url' => 'https://example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'http',
+				),
+				'/c/example.com/path/to/resource.ext',
+			),
+			'no_scheme_content_fail' => array(
+				array(
+					'url' => '//example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => null,
+				),
+				false,
+			),
+			'no_host_scheme_http_fail' => array(
+				array(
+					'url' => '/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'http',
+				),
+				false,
+			),
+			'no_host_scheme_https_fail' => array(
+				array(
+					'url' => '/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'https',
+				),
+				false,
+			),
+			'content_scheme_http_success' => array(
+				array(
+					'url' => '//example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'http',
+				),
+				'/c/example.com/path/to/resource.ext',
+			),
+			'content_scheme_https_success' => array(
+				array(
+					'url' => '//example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'https',
+				),
+				'/c/s/example.com/path/to/resource.ext',
+			),
+			'content_scheme_bad_fail' => array(
+				array(
+					'url' => '//example.com/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => 'bad',
+				),
+				false,
+			),
+			'content_bad_fail' => array(
+				array(
+					'url' => '//example.com/path/to/resource.ext',
+					'content_type' => 'zzz',
+					'scheme' => 'https',
+				),
+				false,
+			),
+			'no_scheme_no_host_fail' => array(
+				array(
+					'url' => '/path/to/resource.ext',
+					'content_type' => 'c',
+					'scheme' => null,
+				),
+				false,
+			),
+			'bad_url_fail' => array(
+				array(
+					'url' => '!@#$%^&*()!@#$%^&*()',
+					'content_type' => 'c',
+					'scheme' => 'http',
+				),
+				false,
+			),
+		);
+	}
+
+	/**
+	 * @dataProvider get_amp_cache_path_for_url_data
+	 * @group amp-cache-path-test	
+	 */
+	public function test_get_amp_cache_path_for_url( $data, $expected_ache_path ) {
+		$cache_path = AMP_Cache_Utilities::get_amp_cache_path_for_url( $data['url'], $data['content_type'] , $data['scheme'] );
+		$this->assertEquals( $cache_path, $expected_ache_path );
+	}
+
+
+	public function get_do_amp_update_ping_data() {
+		return array(
+			'post_success' => array(
+				array(
+					'factory_type' => 'post',
+					'http_response_code' => 204,
+				),
+				true,
+			),
+			'tag_fail' => array(
+				array(
+					'factory_type' => 'tag',
+					'http_response_code' => 204,
+				),
+				false,
+			),
+			'user_fail' => array(
```

Comment:
Yep, I just wanted to make sure that they would be rejected correctly by the function and not actually pinged if someone tried to use this code to cache them. Maybe I should have called this test `reject_user_post_type_success` to be more explicit about what I'm trying to accomplish.